frontend/src/pages/Contracts/Contract/ContractDetails/ContractDetailsSection/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { Box, Button } from "@mui/material";
+import { EditForm } from "components/EditForm";
+import { ReadForm } from "components/ReadForm";
+import { Renderer } from "components/Renderer";
+import { useFormSubmit } from "hooks/useFormSubmit";
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { editFields, readFields } from "./fields";
+
+export const ContractDetailsSection = ({
+  query,
+  userHasEditCapability,
+}: {
+  /**
+   * @todo Define a good type. "Any" type temporarily permitted.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  query: any;
+  userHasEditCapability: boolean;
+}) => {
+  const { contractId } = useParams();
+  const [editMode, setEditMode] = useState(false);
+  const { handleUpdate, Notification } = useFormSubmit();
+
+  useEffect(() => {}, [query]);
+
+  let content = <></>;
```

Comment:
I feel this content conditional, readfrom, editfrom, renderer and notification could be one component, similar to my tableData something like OnPageData or InlineData not sure of the semantics, that way we pass some props, and it handles all this other stuff.

Reply:
I was thinking about that too. I can make it reusable like TableData since this structure is also used in other pages like projects.